### ISSUE TITLE
Add traits for common Host integration tasks

### ISSUE THREAD

**Issue by foundrytom** [OpenAssetIO/OpenAssetIO-MediaCreation#22]

**Comment by feltech** [comment]
Could we sneak in the change from `location` -> `url` as part of this?

**Comment by foundrytom** [comment]
> Could we sneak in the change from `location` -> `url` as part of this?

I don't think so, untill we have a versioning mechanism, as this would break every single integration to date.
